platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/signed/SavedStateMetadata.java

```diff
@@ -314,19 +425,29 @@ private static Instant parseInstant(
     }
 
     /**
-     * Attempt to parse a NodeId from the data map.
+     * Attempt to parse a primitive instant from the data map. Throws if field can't be found or if the value is not a
+     * valid instant.
      *
-     * @param data the data map
-     * @return the parsed NodeId, or null if the field is not present or the value is not a valid Long
+     * @param data  the data map
+     * @param field the field to parse
+     * @return the parsed instant
      */
-    @Nullable
-    private static NodeId parseNodeId(@NonNull final Map<SavedStateMetadataField, String> data) {
-        Objects.requireNonNull(data, "data must not be null");
-        final Long longValue = parseLong(data, SavedStateMetadataField.NODE_ID);
-        if (longValue == null) {
+    @NonNull
+    private static Instant parseNonNullInstant(
+            final Map<SavedStateMetadataField, String> data, final SavedStateMetadataField field) throws IOException {
```

Comment:
missing nullity markings on f the method arguments.

Reply:
added